arlas-core/src/main/java/io/arlas/server/services/ExploreServices.java

```diff
@@ -394,71 +405,101 @@ public AggregationResponse formatAggregationResult(MultiBucketsAggregation aggre
                                 .map(hitsArray -> Arrays.asList(hitsArray))
                                 .map(hitsList -> hitsList.stream().map(hit -> hit.getSourceAsMap()).collect(Collectors.toList()))
                                 .orElse(new ArrayList());
-                    } else if (subAggregation.getName().equals(FluidSearch.DATEHISTOGRAM_AGG) || subAggregation.getName().startsWith(FluidSearch.GEOHASH_AGG) || subAggregation.getName().equals(FluidSearch.HISTOGRAM_AGG) || subAggregation.getName().equals(FluidSearch.TERM_AGG)) {
-                        subAggregationResponse = formatAggregationResult(((MultiBucketsAggregation) subAggregation), subAggregationResponse, collection);
-                    } else if (subAggregationResponse.name.equals(FluidSearch.FIRST_GEOMETRY) || subAggregationResponse.name.equals(FluidSearch.LAST_GEOMETRY) || subAggregationResponse.name.equals(FluidSearch.RANDOM_GEOMETRY)) {
+                    } else if (subAggregation.getName().equals(FluidSearch.DATEHISTOGRAM_AGG) || subAggregation.getName().equals(FluidSearch.GEOHASH_AGG) || subAggregation.getName().equals(FluidSearch.HISTOGRAM_AGG) || subAggregation.getName().equals(FluidSearch.TERM_AGG)) {
```

Comment:
to avoid to check multiple equalities, you could do something like Arrays.asList(FluidSearch.DATEHISTOGRAM_AGG,FluidSearch.GEOHASH_AGG, ... ).contains(subAggregation.getName())

Reply:
fixed

Reply:
fixed